iec61850serversimulator/src/main/java/com/cgi/iec61850serversimulator/functionclass/DatabaseUtils.java

```diff
@@ -234,17 +234,20 @@ public void updateDatabaseSchedule(Schedule schedule) {
 
         int indexNumber = schedule.getIndexNumber();
         int relayNr = schedule.getRelayNr();
-        boolean found = this.scheduleRepository.existsByIndexNumberAndRelayId(schedule.getIndexNumber(), relayNr);
+        List<RelayEntity> relayEntities = this.relayRepository.findByIndexNumber(relayNr);
+        RelayEntity relayEntity = relayEntities.get(0);
+        long relayId = relayEntity.getId();
+        boolean found = this.scheduleRepository.existsByIndexNumberAndRelayId(schedule.getIndexNumber(), relayId);
 
         if (found) {
             List<ScheduleEntity> scheduleEntities = this.scheduleRepository.findByIndexNumberAndRelayId(indexNumber,
-                    relayNr);
+                    relayId);
```

Comment:
Haven't you got this code (or almost exactly the same) somewhere else? If so, turn it into a function.

Reply:
Done by having findRelayIdByRelayNr to not have to initialise the RelayEntity list every time to obtain its relayId.